### PR TITLE
Fixing `arcs_kt_gen` dependency system for G3

### PR DESCRIPTION
**Pull request by alxrsngrtn**
This PR is a pre-requisite for #6039 
- Pulling in the `_manifest` suffix dep into the macro.
- Adding filters / workarounds (to `deps` and `data`) to not break current rules in G3.
- Once G3 rules are modified, a lot of these changes can be modified or removed. 

Evidence that these rules do not break G3: cl/329401762

CC: @csilvestrini

### CODE REVIEW

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 1]
```diff
@@ -738,11 +738,17 @@ def arcs_kt_gen(
     schema_name = name + "_schema"
     plan_name = name + "_plan"
 
+    manifest_deps = [
+        d + "_manifest"
+        for d in manifest_only(deps, inverse = True)
+        if (d.find(":") != -1 and not d.endswith("_manifest"))
```
Comment (csilvestrini) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl:744]: `d.find(":") != -1` could be written more simply: `d[0] != ":"`

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 2]
```diff
@@ -738,11 +738,17 @@ def arcs_kt_gen(
     schema_name = name + "_schema"
     plan_name = name + "_plan"
 
+    manifest_deps = [
```
Comment (csilvestrini) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl:741]: A comment here would be helpful. The logic in this list comprehension is fairly complex

**Review by csilvestrini — APPROVED** [review]

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 2 above)
Comment (bgogul) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:741]: +1

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 1 above)
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:744]: It won't always be the first symbol. 

I originally had `d.startswith(":")`, but it failed in certain cases.

Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:744]: Ah right, I misunderstood the logic for `find`

This is now my new suggestion: `if ":" in d and not ...`

Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:744]: I'm happy with that suggestion, I'll apply the change tomorrow :)

**Pull request merged**